src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -409,6 +408,22 @@ internal string CreateSearchQuery(string[] fields, string SearchTerm, bool IsPac
 
                 if (searchTerm.Contains(' '))
```

Comment:
Can we somehow combine this and the `hasEmptySpaces ` flag, so that we do not have to check this twice, and probably avoid some confusion in the future?

Reply:
I will suggest to consider this refactoring for a different Jira task since I remember it generated other problems also the hasEmptySpaces is under a condition that you added probably we will need to think more about it.
![image](https://github.com/user-attachments/assets/1433ca33-b629-4c9a-a650-4b86529eeb20)

https://github.com/DynamoDS/Dynamo/pull/15338/files